blog/_posts/2016-10-06-julia-0.5-highlights.md

```diff
@@ -0,0 +1,719 @@
+---
+layout: post
+title:  Julia 0.5 Highlights
+author: Stefan Karpinski
+---
+
+Julia 0.5 is a pivotal release.
+It introduces more transformative features than any release since the first official version.
+Moreover, several of these features set the stage for even more to come in the [lead up to Julia 1.0](https://www.youtube.com/watch?v=5gXMpbY1kJY).
+In this post, we'll go through some of the major changes in 0.5, including improvements to functional programming, comprehensions, generators, arrays, strings, and more.
+
+## Functions
+
+Julia has always had first-class functions that can be passed to and from other functions (i.e. [higher-order functions](https://en.wikipedia.org/wiki/Higher-order_function)) and full support for [lambdas](https://en.wikipedia.org/wiki/Anonymous_function).
+Before this release, however, anonymous and higher-order functions came with a significant performance cost, and in a language that targets high-performance technical computing, that's a serious limitation.
+So the Julia standard library and ecosystem have been rife with work-arounds to get the expressiveness of functional programming without the performance problems.
+But the right solution, of course, is to make functional programming fast – ideally just as fast as the optimal hand-written version of your code would be.
+In Julia 0.5, it is.
+And that changes everything.
+
+This change is so important that there will be a separate blog post about it in the coming weeks, explaining how higher-order functions and lambdas have been made so efficient, as well as detailing the kinds of zero-cost abstractions these changes enable.
+But for now, I'll just tease with a little timing comparison.
+First, some definitions – they're the same in both 0.4 and 0.5:
+
+    v = rand(10^7);                   # 10 million random numbers
+    double_it_vec(v) = 2v             # vectorized doubling of input
+    double_it_map(v) = map(x->2x, v)  # map a lambda over input
+
+First, a timing comparison in Julia 0.4:
+
+    julia> VERSION
+    v"0.4.7"
+
+    julia> mean([@elapsed(double_it_vec(v)) for _=1:100])
+    0.024444888209999998
+
+    julia> mean([@elapsed(double_it_map(v)) for _=1:100])
```

Comment:
~~It might be overkill, but I would use `@benchmark` from BenchmarkTools (just to promote people to use that).
The current version is nice in that you can just past it into the REPL.~~ I just read down to the example where you are using BenchmarkTools

Otherwise to improve readability I would put spaces in between the brackets and the content of the comprehension.

`mean([ @elapsed(double_it_map(v)) for _=1:100 ])`
 but syntax highlighting might help readability to make that unnecessary.